File: 2019/01/python/01.py
#!/usr/bin/env python

def fuel_single(mass):
    fuel = (mass//3) - 2
    return(fuel)

def fuel_for_fuel(mass):
    fuel_list = []
    while mass > 0:
        mass = fuel_single(mass)
        if mass <= 0: break
        fuel_list.append(mass)
        
    return(sum(fuel_list))

if __name__ == "__main__":
    with open('../input.txt', 'r') as f:
        input_list = map(int, f.readlines())
        input_list = list(input_list)
    
    p1 = sum(map(fuel_single, input_list))
    p2 = sum(map(fuel_for_fuel, input_list))

    print(f"Part 1 answer: {p1}")
    print(f"Part 2 answer: {p2}")

File: 2019/01/ocaml/01.ml
#use "topfind"
#thread
#require "core.top"
#require "core.syntax"
#load "str.cma"

open Core

let () = 
  let input  = In_channel.read_lines "../input.txt" |> 
                 List.map ~f:int_of_string 
  in

  let p1_ans = input |> 
                 List.map ~f:(fun x -> (x / 3) - 2) |> 
                 List.fold_left ~f:(+) ~init:0 
  in

  printf "Part 1 answer: %d\n" p1_ans;
